WordPress/gutenberg
# Search Block: Add button, label, and width options

Fixes #22071 (partially)
Closes #24643

## Description
Update the search block to support:

* Hiding the search label.
* Switching between a button with text, or an icon.
* Changing the position of the button (outside, inside, button only, no button)
* Resizing the width of the search field.

## How has this been tested?
Tested locally, all e2e tests passed.




## Screenshots 

![2020-09-02 11 19 19](https://user-images.githubusercontent.com/1464705/92021211-4ae3f580-ed0e-11ea-81e3-9282c460164b.gif)


## Types of changes
New feature. I have tested upgrading from the previous version of the block, and all settings and visuals remain consistent.

## Testing Instructions
* Insert a search block into a post or page
* Adjust the settings for the block using the block toolbar
* Confirm these settings are saved, and the search block is rendered correctly on the front end.
* Try adding a search block using master, and then switch to this branch and confirm the block still works and looks correct.

## Checklist:
